### PR TITLE
test-case: manual: add regression test case for linux#1762

### PR DESCRIPTION
**Pull request by ClarexZhou**
add regression test case for linux#1762 : Ensure system not hang when sof probe fail

### CODE REVIEW

File: test-case/manual/regresstion-linux1762-system-hang-when-sof-probe-fail.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+# Preconditions
+PulseAudio should be enabled
+
+# Test Description
+* Remove fw file to let sof probe fail
+
+## Test Case:
+1. Remove fw file, via "sudo mv /lib/firmware/intel/sof /lib/firmware/intel/sof_backup"
+2. Reboot system  -> Ensure system not hang after bootup
+3. Let system enter suspend & wakeup via "sudo rtcwake -m mem -s 10" ->Ensure system not hang
```
Comment (aiChaoSONG) [inline test-case/manual/regresstion-linux1762-system-hang-when-sof-probe-fail.md]: there should be a SPACE after "->"

File: test-case/manual/regresstion-linux1762-system-hang-when-sof-probe-fail.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+4. Recover fw file via "sudo mv /lib/firmware/intel/sof_backup /lib/firmware/intel/sof"
```
Comment (aiChaoSONG) [inline test-case/manual/regresstion-linux1762-system-hang-when-sof-probe-fail.md]: Here use Restore, not Recover

File: test-case/manual/regresstion-linux1762-system-hang-when-sof-probe-fail.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+5. Remove fw-tplg file, via "sudo mv /lib/firmware/intel/sof-tplg /lib/firm/intel/sof-tplg_backup"
+6. Reboot system -> Ensure sytem not hang
+7. "sudo rtcwake -m mem -s 10" -> Ensure system not hang
+8. Recover fw-tplg file
```
Comment (aiChaoSONG) [inline test-case/manual/regresstion-linux1762-system-hang-when-sof-probe-fail.md]: Recover -> Restore

File: test-case/manual/regresstion-linux1762-system-hang-when-sof-probe-fail.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+# Preconditions
+PulseAudio should be enabled
+
+# Test Description
+* Remove fw file to let sof probe fail
+
+## Test Case:
+1. Remove fw file, via "sudo mv /lib/firmware/intel/sof /lib/firmware/intel/sof_backup"
+2. Reboot system  -> Ensure system not hang after bootup
```
Comment (aiChaoSONG) [inline test-case/manual/regresstion-linux1762-system-hang-when-sof-probe-fail.md]: Two spaces before "->" , but one space after "->", please remove one before "->"

File: test-case/manual/regresstion-linux1762-system-hang-when-sof-probe-fail.md [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+3. Let system enter suspend & wakeup via "sudo rtcwake -m mem -s 10" ->Ensure system not hang
+4. Recover fw file via "sudo mv /lib/firmware/intel/sof_backup /lib/firmware/intel/sof"
+5. Remove fw-tplg file, via "sudo mv /lib/firmware/intel/sof-tplg /lib/firm/intel/sof-tplg_backup"
+6. Reboot system -> Ensure sytem not hang
+7. "sudo rtcwake -m mem -s 10" -> Ensure system not hang
+8. Recover fw-tplg file 
+9. Reboot system -> sof is loaded sucessfully.
+10. Remove fw file
+11. "sudo rtcwake -m mem -s 10" -> Ensure system not hang.
+12. Recover fw file and reboot -> sof is loaded sucessfully
```
Comment (aiChaoSONG) [inline test-case/manual/regresstion-linux1762-system-hang-when-sof-probe-fail.md]: Recover -> Restore

File: test-case/manual/regresstion-linux1762-system-hang-when-sof-probe-fail.md [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+13. "sudo rmmmod snd_sof_pci" -> Ensure system not hang
```
Comment (aiChaoSONG) [inline test-case/manual/regresstion-linux1762-system-hang-when-sof-probe-fail.md]: rmmmod -> rmmod

**Comment by aiChaoSONG** [comment]
@fredoh9 @stripes416 Please help to review this, if this looks good to you, let's get it merged.

File: test-case/manual/regresstion-linux1762-system-hang-when-sof-probe-fail.md [hunk 7]
```diff
@@ -0,0 +1,28 @@
+# Linux#1762 Regression test
+Ensure system not hang when sof probe failed
+
+# Preconditions
+PulseAudio should be enabled
+
+# Test Description
+* Remove fw file to let sof probe fail
```
Comment (xiulipan) [inline test-case/manual/regresstion-linux1762-system-hang-when-sof-probe-fail.md]: I think you also remove tplg to make tplg load fail?

File: test-case/manual/check-fail-no-tplg.md [hunk 8]
```diff
@@ -0,0 +1,22 @@
+# No sof tplg test
+Ensure system not hang when missing sof-tplg
```
Comment (stripes416) [inline test-case/manual/check-fail-no-tplg.md]:
```suggestion
The missing SOF tplg test is a regression test based on a sof_probe function failure documented in [linux#1762](https://github.com/thesofproject/linux/issues/1762) and is part of a series of manual tests associated with regression-linux1762-system-hand-when-sof-probe-fail.md.
```

File: test-case/manual/check-fail-no-tplg.md [hunk 9]
```diff
@@ -0,0 +1,22 @@
+# No sof tplg test
+Ensure system not hang when missing sof-tplg
+
+# Preconditions
+PulseAudio should be enabled
+
+# Test Description
+* Remove sof-tplg file to let sof probe fail
```
Comment (stripes416) [inline test-case/manual/check-fail-no-tplg.md]:
```suggestion
* Check system does not hang when expected SOF tplg is missing by renaming the tplg directory and letting sof_probe functions fail
```

File: test-case/manual/check-fail-no-tplg.md [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+2. Reboot system -> Ensure sytem not hang and report missing sof-tplg error in dmesg
+3. Let system enter suspend then wakeup, via "sudo rtcwake -m mem -s 10" -> Ensure system not hang
+4. Restore sof-tplg file, via "sudo mv /lib/firmware/intel/sof-tplg_backup /lib/firmware/intel/sof-tplg"
+5. Reboot system -> sof is loaded sucessfully.
+6. Remove sof-tplg file
+7. "sudo rtcwake -m mem -s 10" -> Ensure system not hang.
+8. Restore sof-tplg file and reboot -> sof is loaded sucessfully
+
+## Expected results
+* System should not hang when missing sof-tplg
```
Comment (stripes416) [inline test-case/manual/check-fail-no-tplg.md]:
```suggestion
* System should not hang when expected SOF tplg is missing
```

File: test-case/manual/check-fail-no-tplg.md [hunk 11]
```diff
@@ -0,0 +1,22 @@
+# No sof tplg test
+Ensure system not hang when missing sof-tplg
+
+# Preconditions
+PulseAudio should be enabled
+
+# Test Description
+* Remove sof-tplg file to let sof probe fail
+
+## Test Case:
```
Comment (stripes416) [inline test-case/manual/check-fail-no-tplg.md]: Same question here again --- do you intend for the manual test to be executed twice or are you testing two similar but slightly different cases? If its just one case executed twice, lets remove the duplication of instructions and just say to run it twice. If its two slightly different cases, lets draw attention to that and make that clearer.

- Instructions 1-5 / Case 1:
  - system is on
  - rename sof-tplg dir
  - reboot
  - check that missing sof-tplg error is in dmesg
  - test s/r cycle
  - restore sof-tplg dir
  - reboot
  - sof-tplg loads properly
<br></br>
- Instructions 6-8 / Case 2 (same as case 1, just no reboot immediately after sof-tplg dir is renamed): 
  - system is on
  - rename sof-tplg dir
  - ~reboot~
  - check that missing sof-tplg error is reported in dmesg
  - test s/r cycle
  - restore sof-tplg dir
  - reboot
  - sof-tplg loads properly

File: test-case/manual/check-fail-no-tplg.md [hunk 12]
```diff
@@ -0,0 +1,22 @@
...
+Ensure system not hang when missing sof-tplg
+
+# Preconditions
+PulseAudio should be enabled
+
+# Test Description
+* Remove sof-tplg file to let sof probe fail
+
+## Test Case:
+1. Remove sof-tplg file, via "sudo mv /lib/firmware/intel/sof-tplg /lib/firmware/intel/sof-tplg_backup"
```
Comment (stripes416) [inline test-case/manual/check-fail-no-tplg.md]:
```suggestion
1. Rename sof-tplg directory, via `sudo mv /lib/firmware/intel/sof-tplg /lib/firmware/intel/sof-tplg_backup`
```

File: test-case/manual/check-fail-no-tplg.md [hunk 13]
```diff
@@ -0,0 +1,22 @@
...
+2. Reboot system -> Ensure sytem not hang and report missing sof-tplg error in dmesg
+3. Let system enter suspend then wakeup, via "sudo rtcwake -m mem -s 10" -> Ensure system not hang
```
Comment (stripes416) [inline test-case/manual/check-fail-no-tplg.md]:
```suggestion
3. Let system enter suspend then wakeup, via `sudo rtcwake -m mem -s 10` -> System should not hang during or after suspend / resume cycle
```

File: test-case/manual/check-fail-no-tplg.md [hunk 14]
```diff
@@ -0,0 +1,22 @@
...
+4. Restore sof-tplg file, via "sudo mv /lib/firmware/intel/sof-tplg_backup /lib/firmware/intel/sof-tplg"
```
Comment (stripes416) [inline test-case/manual/check-fail-no-tplg.md]:
```suggestion
4. Restore sof-tplg directory, via `sudo mv /lib/firmware/intel/sof-tplg_backup /lib/firmware/intel/sof-tplg`
```

File: test-case/manual/check-fail-no-tplg.md [hunk 15]
```diff
@@ -0,0 +1,22 @@
...
+5. Reboot system -> sof is loaded sucessfully.
```
Comment (stripes416) [inline test-case/manual/check-fail-no-tplg.md]:
```suggestion
5. Reboot system -> SOF should be loaded successfully
```

File: test-case/manual/check-fail-no-tplg.md [hunk 16]
```diff
@@ -0,0 +1,22 @@
...
+
+# Preconditions
+PulseAudio should be enabled
+
+# Test Description
+* Remove sof-tplg file to let sof probe fail
+
+## Test Case:
+1. Remove sof-tplg file, via "sudo mv /lib/firmware/intel/sof-tplg /lib/firmware/intel/sof-tplg_backup"
+2. Reboot system -> Ensure sytem not hang and report missing sof-tplg error in dmesg
```
Comment (stripes416) [inline test-case/manual/check-fail-no-tplg.md]:
```suggestion
2. Reboot system -> System should not hang after bootup and missing sof-tplg error is reported in dmesg

```

File: test-case/manual/check-fail-no-tplg.md [hunk 17]
```diff
@@ -0,0 +1,22 @@
...
+3. Let system enter suspend then wakeup, via "sudo rtcwake -m mem -s 10" -> Ensure system not hang
+4. Restore sof-tplg file, via "sudo mv /lib/firmware/intel/sof-tplg_backup /lib/firmware/intel/sof-tplg"
+5. Reboot system -> sof is loaded sucessfully.
+6. Remove sof-tplg file
```
Comment (stripes416) [inline test-case/manual/check-fail-no-tplg.md]:
```suggestion
6. Rename sof-tplg directory, via `sudo mv /lib/firmware/intel/sof-tplg /lib/firmware/intel/sof-tplg_backup`
```

File: test-case/manual/check-fail-no-tplg.md [hunk 18]
```diff
@@ -0,0 +1,22 @@
...
+7. "sudo rtcwake -m mem -s 10" -> Ensure system not hang.
```
Comment (stripes416) [inline test-case/manual/check-fail-no-tplg.md]:
```suggestion
7. Let system enter suspend then wakeup, via `sudo rtcwake -m mem -s 10` -> System should not hang during or after suspend / resume cycle
```

File: test-case/manual/check-fail-no-tplg.md [hunk 19]
```diff
@@ -0,0 +1,22 @@
...
+8. Restore sof-tplg file and reboot -> sof is loaded sucessfully
```
Comment (stripes416) [inline test-case/manual/check-fail-no-tplg.md]:
```suggestion
8. Restore sof-tplg directory, via `sudo mv /lib/firmware/intel/sof-tplg_backup /lib/firmware/intel/sof-tplg`
9. Reboot system -> SOF should be loaded successfully
```

File: test-case/manual/regresstion-linux1762-system-hang-when-sof-probe-fail.md [hunk 20]
```diff
@@ -0,0 +1,18 @@
+# Linux#1762 Regression test
+Ensure system not hang when sof probe failed
```
Comment (stripes416) [inline test-case/manual/regresstion-linux1762-system-hang-when-sof-probe-fail.md]:
```suggestion
The missing SOF firmware and tplg tests are regression tests based on a sof_probe function failure documented in [linux#1762](https://github.com/thesofproject/linux/issues/1762).
```

File: test-case/manual/regresstion-linux1762-system-hang-when-sof-probe-fail.md [hunk 21]
```diff
@@ -0,0 +1,18 @@
...
+# Preconditions
+PulseAudio should be enabled
+
+# Test Description
+* Remove fw file to let sof probe fail
+
+## Test Case:
+1. Test check-fail-no-fw.md
+2. Test check-fail-no-tplg.md
+3. "sudo rmmod snd_sof_pci" -> Ensure system not hang
```
Comment (stripes416) [inline test-case/manual/regresstion-linux1762-system-hang-when-sof-probe-fail.md]:
```suggestion
3. `sudo rmmod snd_sof_pci` -> System should not hang
```

File: test-case/manual/regresstion-linux1762-system-hang-when-sof-probe-fail.md [hunk 22]
```diff
@@ -0,0 +1,18 @@
...
+4. "sudo rtcwake -m mem -s 10" -> Ensure system not hang.
```
Comment (stripes416) [inline test-case/manual/regresstion-linux1762-system-hang-when-sof-probe-fail.md]:
```suggestion
4. `sudo rtcwake -m mem -s 10` -> System should not hang
```

File: test-case/manual/regresstion-linux1762-system-hang-when-sof-probe-fail.md [hunk 23]
```diff
@@ -0,0 +1,18 @@
+# Linux#1762 Regression test
+Ensure system not hang when sof probe failed
+
+# Preconditions
+PulseAudio should be enabled
+
+# Test Description
+* Remove fw file to let sof probe fail
```
Comment (stripes416) [inline test-case/manual/regresstion-linux1762-system-hang-when-sof-probe-fail.md]:
```suggestion
* Check that system does not hang when expected SOF fw and tplg are missing and letting sof_probe functions fail. Once they are restored, check to make sure all is working properly.
```

File: test-case/manual/regresstion-linux1762-system-hang-when-sof-probe-fail.md [hunk 24]
```diff
@@ -0,0 +1,18 @@
...
+
+## Test Case:
+1. Test check-fail-no-fw.md
+2. Test check-fail-no-tplg.md
+3. "sudo rmmod snd_sof_pci" -> Ensure system not hang
+4. "sudo rtcwake -m mem -s 10" -> Ensure system not hang.
+
+## Expected results
+* System should not hang when sof probe fail.
```
Comment (stripes416) [inline test-case/manual/regresstion-linux1762-system-hang-when-sof-probe-fail.md]:
```suggestion
* System should not hang when expected SOF fw and tplg are missing
```

**Pull request merged**